ios/RNSVGRenderable.m

```diff
@@ -187,6 +189,9 @@ - (void)renderLayerTo:(CGContextRef)context rect:(CGRect)rect
         self.path = CGPathRetain(CFAutorelease(CGPathCreateCopy([self getPath:context])));
         [self setHitArea:self.path];
     }
+    
+    const CGRect pathBouding = CGPathGetBoundingBox(self.path);
+    self.clientRect = CGRectApplyAffineTransform(pathBouding, [self.svgView getReverseTransform]);
```

Comment:
The CGContext already has the CTM no need to recompute over and over.

Reply:
That was my initial thought.

`self.clientRect = CGRectApplyAffineTransform(pathBouding, CGAffineTransformInvert(CGContextGetCTM(context)));`

what can be easier, huh?
But no, that's what it gives:

![simulator screen shot - iphone 6s - 2018-06-07 at 23 25 22](https://user-images.githubusercontent.com/7002833/41124294-32ba0334-6aaa-11e8-89a4-bf51085cd233.png)

instead of

![simulator screen shot - iphone 6s - 2018-06-07 at 23 26 06](https://user-images.githubusercontent.com/7002833/41124301-36c0faaa-6aaa-11e8-9f14-397e72c52403.png)

It is very much appreciated if you can suggest better formula for properly converting coordinate from svg space to view

Reply:
One more thing is to be aware is that initial ctm for the context *is not* identity matrix.
I also tried to save that initial CTM and multiply it on the left final ctm.

sort of: `reversetransform = invert(invert(initialTransform) * currentCTM)` and variations of that

Reply:
`viewTransform = initialCTM * viewBoxCTM * nodeTransform_1 * ... * nodeTransform_n`

I thought I need `invert(viewBoxCTM * nodeTransform_1 * ... * nodeTransform_n)`

I tried `reverseTransform = invert(invert(initialCTM) * initialCTM * viewBoxCTM * .... *)`

am I missing something in how matrix multiplications work?

Reply:
Hmm, I don't see you inverting the transforms in the current code, can it be that?

Reply:
That's the point, I couldn't achieve desired behavior with matrix operations. The final code you see was rather a wild guess.

Reply:
as I said I was trying to distill view coordinates through inversions, however the *reverese* multiplication of matrices worked, not the *inversion*

Reply:
Or, if the initial CTM is not the identity matrix, then you might want invert(initial) * CTM, as if CTM = initial * viewbox * elementTMs then inverting the initial one and multiplying from the same direction should get it (close to, still floats involved) identity.

Reply:
I tried.

`reverse = ^(^initial * ctm) = ^(^initial * initial * vb * elements) = ^(identity * vb * elements) = ^(vb * elements)`

the code below:

```
    const CGAffineTransform ctm = CGContextGetCTM(context);
    const CGAffineTransform invertIntial = CGAffineTransformInvert(self.svgView.initialTransform);
    self.clientRect = CGRectApplyAffineTransform(pathBouding, CGAffineTransformInvert(CGAffineTransformConcat(invertIntial, ctm)));
```

doesn't work

Reply:
Well, matrix multiplication is associative, should it shouldn't matter in what order you do the multiplication (reverse, in order, or, some order which minimises the number of scalar multiplications needed, although with same size matrices that doesn't even vary). So (A*B)*C=A*(B*C) but it's not commutative, so in general A*B != B*A

Reply:
ok actually,
this:
```
    const CGAffineTransform ctm = CGContextGetCTM(context);
    const CGAffineTransform invertIntial = CGAffineTransformInvert(self.svgView.initialTransform);
    self.clientRect = CGRectApplyAffineTransform(pathBouding, CGAffineTransformConcat(ctm, invertIntial));
```
works..

but its rather

`reverse = ctm * ^initial`

how can this be right?

Reply:
Seems you have an extra invert there, how about
```
    const CGAffineTransform ctm = CGContextGetCTM(context);
    const CGAffineTransform invertIntial = CGAffineTransformInvert(self.svgView.initialTransform);
    self.clientRect = CGRectApplyAffineTransform(pathBouding, CGAffineTransformConcat(invertIntial, ctm));
```

Reply:
oh wow... damn it!
from docs of `CGContextConcatCTM`:

> CTMnew = transform * CTMcontext.

so basically matrices are added on the left!

`ctm = element_n * ... * element_1 * vb * initial`

then `reverse = ctm * ^initial = element_n * ... * element_1 * vb * initial * ^initial = element_n * ... * element_1 * vb * identity = element_n * ... * element_1 * vb`

but there's still no *inversion* of pure transform

I'm confused %)

I'll update PR later, thank you very much for your comments.

Reply:
Or can the initial transform be due to the viewbox & align?

Reply:
Oh great, thanks for your work as well! Seems great to have onLayout available.